ipv8/test/REST/test_attestation_endpoint.py

```diff
@@ -0,0 +1,477 @@
+from base64 import b64encode
+from json import dumps
+from twisted.internet import reactor
+from twisted.internet.defer import returnValue, inlineCallbacks
+from twisted.internet.task import deferLater
+
+from ..mocking.rest.base import RESTTestBase
+from ..mocking.rest.peer_interactive_behavior import AndroidTestPeer
+from ..mocking.rest.rest_peer_communication import string_to_url
+from ..mocking.rest.rest_api_peer import TestPeer
+
+
+class TestAttestationEndpoint(RESTTestBase):
+    """
+    Class for testing the REST API of the IPv8 object
+    """
+
+    def setUp(self):
+        super(TestAttestationEndpoint, self).setUp()
+        self.initialize([(1, TestPeer)])
+
+    @inlineCallbacks
+    def wait_for_peers(self, dict_param, excluded_peer_mids=None):
+        """
+        Wait until this peer receives a non-empty list of fellow peers in the network
+
+        :param dict_param: the required parameters by the GET request generator for the peers request type
+        :param excluded_peer_mids: A list of peer mids which should not be taken into consideration peers
+        :return: a list of currently known peers in the network
+        """
+        assert isinstance(excluded_peer_mids, (list, set)) or not excluded_peer_mids, "excluded_peer_mids " \
+                                                                                      "must be a list, set or None"
+
+        # Make sure excluded_peer_mids is a set
+        if not excluded_peer_mids:
+            excluded_peer_mids = set()
+        elif isinstance(excluded_peer_mids, list):
+            excluded_peer_mids = set(excluded_peer_mids)
+
+        peer_list = yield self._get_style_requests.make_peers(dict_param)
+        peer_list = set(peer_list)
+
+        # Keep iterating until peer_list is non-empty
+        while not peer_list - excluded_peer_mids:
+            yield deferLater(reactor, 0.1, lambda: None)
```

Comment:
This is a bit ugly, please refactor this into a `yield self.sleep(0.1)` call (defined in `RESTTestBase`).